pkg/prometheus/promcfg.go

```diff
@@ -623,6 +628,26 @@ func (cg *ConfigGenerator) Generate(
 	return yaml.Marshal(cfg)
 }
 
+func (cg *ConfigGenerator) appendStorageSettingsConfig(cfg yaml.MapSlice, p *v1.Prometheus) (yaml.MapSlice, error) {
+	if p.Spec.Exemplars == nil || p.Spec.Exemplars.MaxSize == nil {
+		return cfg, nil
+	}
+
+	cfg = cg.WithMinimumVersion("2.29.0").AppendMapItem(cfg, "storage", yaml.MapSlice{
+		{
+			Key: "exemplars",
+			Value: yaml.MapSlice{
+				{
+					Key:   "max_exemplars",
+					Value: p.Spec.Exemplars.MaxSize,
```

Comment:
(nit) not really required but more explicit

```suggestion
					Value: *p.Spec.Exemplars.MaxSize,
```

Reply:
done :)